connector/apmconnector/metric_to_metric_connector.go

```diff
@@ -0,0 +1,257 @@
+				}
+
+				if f > 0 {
+					generateApdexMetrics(apdex, "F", metrics, dp.StartTimestamp(), dp.Timestamp(), int64(f), name)
+				}
+			}
+		}
+		newMetric.Histogram().SetAggregationTemporality(m.Histogram().AggregationTemporality())
+		newMetric.CopyTo(smNew.Metrics().AppendEmpty())
+	case pmetric.MetricTypeExponentialHistogram:
+		newMetric.SetEmptyExponentialHistogram().DataPoints().EnsureCapacity(3)
+		for i := 0; i < m.ExponentialHistogram().DataPoints().Len(); i++ {
+			dp := m.ExponentialHistogram().DataPoints().At(i)
+			newDp := newMetric.ExponentialHistogram().DataPoints().AppendEmpty()
+			dp.CopyTo(newDp)
+			name, txType := GetTransactionMetricNameFromAttributes(dp.Attributes())
+			newDp.Attributes().Clear()
+			newDp.Attributes().PutStr("transactionType", txType.AsString())
+			newDp.Attributes().PutStr("transactionName", name)
+			newDp.Attributes().PutStr("metricTimesliceName", name)
+
+			isError := ContainsErrorHTTPStatusCode(dp.Attributes())
+			if isError {
+				{
+					attributes := pcommon.NewMap()
+					attributes.PutStr("transactionType", txType.AsString())
+					sum := metrics.GetSum("apm.service.error.count", attributes, true, dp.StartTimestamp(), dp.Timestamp())
+					sum.Add(int64(newDp.Count()), dp.StartTimestamp(), dp.Timestamp())
+				}
+				{
+					attributes := pcommon.NewMap()
+					attributes.PutStr("transactionType", txType.AsString())
+					attributes.PutStr("transactionName", name)
+					sum := metrics.GetSum("apm.service.transaction.error.count", attributes, true, dp.StartTimestamp(), dp.Timestamp())
+					sum.Add(int64(newDp.Count()), dp.StartTimestamp(), dp.Timestamp())
+				}
+
+				generateApdexMetrics(apdex, "F", metrics, dp.StartTimestamp(), dp.Timestamp(), int64(dp.Count()), name)
+			}
+
+			// TODO: Generate apdex metrics for exponential histograms.
+		}
+		newMetric.ExponentialHistogram().SetAggregationTemporality(m.ExponentialHistogram().AggregationTemporality())
+		newMetric.CopyTo(smNew.Metrics().AppendEmpty())
+	default:
+		logger.Error("unexpected metric type", zap.String("name", m.Name()), zap.String("type", metricType.String()))
+	}
+}
+
+func createResourceAndScopeMetrics(logger *zap.Logger, rmNew pmetric.ResourceMetrics, attributesFilter *AttributeFilter, rm pmetric.ResourceMetrics, newMetrics pmetric.Metrics, metrics *ResourceMetrics, metricMap Metrics, smNew pmetric.ScopeMetrics) (pmetric.ResourceMetrics, pmetric.ScopeMetrics, *ResourceMetrics) {
+	if rmNew == (pmetric.ResourceMetrics{}) {
+		resourceAttributes, err := attributesFilter.FilterAttributes(rm.Resource().Attributes())
+		if err != nil {
+			logger.Error("Could not filter resource attributes", zap.String("error", err.Error()))
+		}
+		rmNew = newMetrics.ResourceMetrics().AppendEmpty()
+		resourceAttributes.CopyTo(rmNew.Resource().Attributes())
+
+		// TODO: should we declare a New Relic specific schema?
+		// rmNew.SetSchemaUrl(rm.SchemaUrl())
```

Comment:
add an issue, maybe? I think it would make sense to publish our schema.

Reply:
I agree we should declare our own schema. Question I had on my mind is where would it live? It is likely at this point that we will move this metric generation to the backend instead of performing it in the collector. If we published a NR schema then all NR products/services that generate data based on the schema should use the common schema url.